src/sc/sidechaintypes.cpp

```diff
@@ -273,10 +274,13 @@ CFieldElement CFieldElement::ComputeHash(const CFieldElement& lhs, const CFieldE
 
 const CFieldElement& CFieldElement::GetPhantomHash()
 {
-    // TODO call an utility method to retrieve from zendoo_mc_cryptolib a constant phantom hash
-    // field element and use it everywhere it is needed a constant value whose preimage has to
-    // be unknown
-    static CFieldElement ret{std::vector<unsigned char>(CFieldElement::ByteSize(), 0x00)};
+    // zendoo_mc_cryptolib allocates a new buffer with static lifetime on each call to zendoo_get_phantom_cert_data_hash().
+    static field_t* phantom_ptr = zendoo_get_phantom_cert_data_hash();
+    size_t size = (int)CFieldElement::ByteSize() / sizeof(unsigned char);
```

Comment:
Why convert sign when operands and result are meant to be unsigned ?

Reply:
Also not very clear to me how _staticness_ is enforced. Apparently at every call a buffer is allocated

Reply:
Woops, this is a leftover from a previous implementation. I updated the function entirely to avoid memcopies on each call, please see 7e39749d9b01b1ee9557e5bc221d95963163f968